website/docs/feature-flag-tutorials/python/examples.md

```diff
@@ -0,0 +1,508 @@
+
+
+## Canary Deployments in Python
+
+
+### What is a canary deployment?
+
+Canary releases are a way to test and release code in different environments for a subset of your audience, which determines which features or versions of the platform people have access to.
+
+
+### How to do canary deployments with a Python feature flag?
+
+
+Canary deployments are a safer and more gradual way to make changes in software development. They help find abnormalities and align with the agile process for faster releases and quick reversions.
+
+Unleash has a few ways to help manage canary deployments for Python apps at scale:
+
+- Using a [gradual rollout](/reference/activation-strategies#gradual-rollout) (which we implemented in a [previous section](#gradual-rollouts-for-python-apps)) would be a simple use case but would reduce the amount of control you have over who gets the new feature.
+
+- Using either [strategy constraints](/reference/strategy-constraints) or [segments](/reference/segments) (which are a collection of constraints) to determine which user receives which version for more control than a gradual rollout
+
+- [Strategy variants](/reference/strategy-variants) are used for more advanced cases. For example, if you have 2+ new features and are testing to see if they are better than the old one, you can use strategy variants to split your population of users and conduct an A/B test with them.
+
+Let’s walk through how to utilize **strategy constraints** in our Python app.
+
+
+### Configure strategy constraints for canary deployments
+
+
+We will build a strategy constraint on our existing gradual rollout strategy. This will allow us to target a subset of users for the rollout.
+
+In Unleash, start from the feature flag view and edit your Gradual Rollout strategy from your development environment.
+
+
+![Edit your gradual rollout strategy by selecting the 'edit strategy' button.](/img/python-ex-edit-strategy.png)
+
+
+This will take you to the gradual rollout form. Next, add a new constraint.
+
+
+![Add a constraint to your strategy.](/img/python-ex-add-constraint.png)
+
+Let’s say we are experimenting with releasing the “delete” feature to all environments except production to test it with internal users before releasing it to all users.
+
+![The new constraint form includes a context field, operator, and values field to customize the conditions under which a user will be exposed to the flag](/img/python-ex-constraint-page.png)
+
+
+We can configure the constraint in the form to match these requirements:
+- The context field is set to **environment**
+- The operator is set to **NOT_IN**
+- The value added is **production**
+
+
+Once you’ve filled out the proper constraint fields, select ‘Done’ and save the strategy.
+
+![Once you add a constraint, you can see it listed underneath your strategy in Unleash.](/img/python-ex-constraint-added.png)
+
+
+Your release process is now configured with an environment-dependent strategy constraint.
```

Comment:
Somewhere in this section it might be worth mentioning that we've set the rollout to 100%, which means 100% of our user base as long as they do not belong to the `production` environment.